source/common/network/resolver_impl.cc

```diff
@@ -54,6 +54,27 @@ InstanceConstSharedPtr resolveProtoAddress(const envoy::api::v2::core::Address&
   }
 }
 
+InstanceConstSharedPtr
+resolveProtoAddress(const envoy::api::v2::core::Address& address,
+                    const envoy::api::v2::Cluster::DiscoveryType& cluster_type) {
+  switch (address.address_case()) {
+  case envoy::api::v2::core::Address::kSocketAddress:
+    try {
+      return resolveProtoSocketAddress(address.socket_address());
+    } catch (EnvoyException& e) {
+      if (cluster_type == envoy::api::v2::Cluster::STATIC) {
```

Comment:
Doesn't this issue also apply to EDS clusters which I think is also where some of the confusion is coming from?

Reply:
Woops, yes, we need to check for `envoy::api::v2::Cluster::EDS`. Will fix, either here or in a helper function in `Upstream` namespace.

Reply:
Thanks. I guess I would vote for helper wrapper in `Upstream` namespace.